src/storage/v2/inmemory/storage.cpp

```diff
@@ -234,7 +235,9 @@ VertexAccessor InMemoryStorage::InMemoryAccessor::CreateVertexEx(storage::Gid gi
 
   auto *delta = CreateDeleteObjectDelta(&transaction_);
   auto [it, inserted] = acc.insert(Vertex{gid, delta});
-  MG_ASSERT(inserted, "The vertex must be inserted here!");
+  if (!inserted) {
+    throw utils::BasicException("The edge must be inserted here");
```

Comment:
+ if we throw, the error message has to be much more descriptive / saying what should user do about this + assist him

in addition, the convention on a storage layer is not to `throw` because of the issue related to exceptions, consider returning `optional/result<VertexAccessor>` instead of plain `VertexAccessor` to force the error handling on the caller side... this is also not ideal because we'll change the storage API (tricky)

CC @antoniofilipovic not sure how critical is this, but we probably need a design sync here 😃

Reply:
It is not critical, but as I described here https://github.com/memgraph/memgraph/pull/1208#issuecomment-1697233546 it started as an idea to support loading JSON with given IDs in file as Neo4j supports it. I agree that instead of error, optional/Result<VertexAccessor> is better than throwing error.